File: tests/java/src/main/java/com/escape/tests/cases/Case030ArchiveLedger.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 030: archive ledger payload for downstream business processing. */
public class Case030ArchiveLedger {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: archive ledger records and prepare transport-ready payload.
        String taskName = "archive_ledger";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "ledger");
        payload.put("stage", "archive");
        payload.put("input", raw);
        // SAFE: payload remains local; only primitive summary string is returned.
        return payload.get("task") + ":" + payload.get("input");
    }
}

File: tests/go/cases/case_030_archive_ledger.go
package escape_tests

var retainedCase030 = []map[string]string{}

func Case030ArchiveLedger(input string) string {
    // Task case 030: archive ledger payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "archive_ledger",
        "entity": "ledger",
        "stage": "archive",
        "input": raw,
    }
    // SAFE: payload remains local; only primitive summary string is returned.
    return payload["task"] + ":" + payload["input"]
}
